posthog/clickhouse/migrations/0106_materialize_session_ids_uuid.py

```diff
@@ -0,0 +1,27 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.client.connection import get_client_from_pool
+from posthog.clickhouse.kafka_engine import trim_quotes_expr
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+
+ADD_COLUMNS_SHARDED_EVENTS = f"""
+ALTER TABLE {{table}} ON CLUSTER {{cluster}}
+ADD COLUMN IF NOT EXISTS $session_id_uuid Nullable(UInt128) MATERIALIZED toUInt128(JSONExtract(properties, '$session_id', 'Nullable(UUID)'))
+"""
+
+ADD_COLUMNS_EVENTS = """
+ALTER TABLE {table} ON CLUSTER {cluster}
+ADD COLUMN IF NOT EXISTS $session_id_uuid Nullable(UInt128)
+"""
+
+
+def add_columns_to_required_tables(_):
+    with get_client_from_pool() as client:
+        client.execute(ADD_COLUMNS_SHARDED_EVENTS.format(table="sharded_events", cluster=CLICKHOUSE_CLUSTER))
+        client.execute(ADD_COLUMNS_EVENTS.format(table="events", cluster=CLICKHOUSE_CLUSTER))
```

Comment:
Can we run these using the `run_sql_with_exceptions` function?

We need to update the distributed table in the coordinators as well. The sharded table only in the data nodes (which is the default).

Something like this (without the need of using `on cluster`).

```suggestion
run_sql_with_exceptions(ADD_COLUMNS_SHARDED_EVENTS.format(table="sharded_events"))
run_sql_with_exceptions(ADD_COLUMNS_EVENTS.format(table="events"), node_role=NodeRole.ALL)
```